hw2/hw2.py

```diff
@@ -0,0 +1,131 @@
+"""Module for processing data."""
+
+
+import json
+import os
+from typing import Callable
+
+from app_exceptions import NotFoundAgeException
+from user_statistic import UserStatistic
+
+
+def exceptions_to_json(function: Callable) -> None:
+    """Convert exceptions to json.
+
+    Args:
+        function (Callable): function.
+
+    Returns:
+        None: returns function-wrapper
+    """
+    def wrapper(*args):
+        try:
+            function(*args)
+        except Exception as error:
+            error_message = json.dumps({'error': str(error)})
+            write_json(args[1], error_message)
+    return wrapper
+
+
+@exceptions_to_json
+def process_data(input_path: str, output_path: str) -> None:
+    """Take data and writes the statictic by age and registrations by years to the json file.
+
+    Args:
+        input_path (str): the path where the json comes from.
+        output_path (str): path where json is saved.
+
+    Raises:
+        NotFoundAgeException: raise exception if we don't find 'age' in users.
+    """
+    users = get_users_from_json(input_path)
+    try:
+        ages = [user['age'] for user in users.values()]
+    except KeyError:
+        raise NotFoundAgeException('The parametr <age> is not found.')
+
+    check_ages_type(ages)
+
+    stats = get_years_statistic(users)
+
+    user_stats = UserStatistic(ages, stats).to_json()
+
+    write_json(output_path, user_stats)
+
+
+def write_json(output: str, message: any) -> None:
+    """Write json with message to output.
+
+    Args:
+        output (str): The place where we save json file.
+        message (any): Some data which we write.
+    """
+    if not os.path.exists(output):
+        os.mkdir('test_data_hw2')
```

Comment:
```suggestion
      os.makedirs('test_data_hw2', exist_ok=True)
```
https://docs-python.ru/standart-library/modul-os-python/funktsija-makedirs-modulja-os/

Reply:
поменял